src/util-file-swf-decompression.c

```diff
@@ -144,46 +140,30 @@ int FileSwfLzmaDecompression(DetectEngineThreadCtx *det_ctx,
 {
     int ret = 0;
 
-    CLzmaDec strm;
-    LzmaDec_Construct(&strm);
-    ELzmaStatus status;
-
-    if (compressed_data_len < LZMA_PROPS_SIZE + 8) {
-        DetectEngineSetEvent(det_ctx, FILE_DECODER_EVENT_LZMA_FORMAT_ERROR);
-        return 0;
-    }
-    ret = LzmaDec_Allocate(&strm, compressed_data, LZMA_PROPS_SIZE, &suri_lzma_Alloc);
-    if (ret != SZ_OK) {
-        DetectEngineSetEvent(det_ctx, FILE_DECODER_EVENT_LZMA_DECODER_ERROR);
-        return 0;
-    }
-    LzmaDec_Init(&strm);
-    compressed_data += LZMA_PROPS_SIZE + 8;
```

Comment:
Why did these got removed ? Did you test this ?

Reply:
This is advancing the input buffer past the header / properties, which were parsed out in `LzmaDec_Allocate()`. lzma-rs manages this internally, and will first read through all of the properties, and then start decompressing the data. 

You can see the relevant part of lzma-rs [here](https://github.com/gendx/lzma-rs/blob/6e1f0d797230ea3c30f9c002967f1151ac555124/src/decode/stream.rs#L228). It starts by parsing the properties in the header, and then reads the rest, all within the `write()` function where you stuff data in.

As for testing, there are several tests in lzma-rs against the short header case and full decompression, suricata has several LZMA tests in `tests/detect-http-server-body.c`, and we have a few lzma decompression tests in libhtp-rs that use the same API. Those tests all pass, so I don't think manually advancing the compression stream past the lzma header is necessary.

Reply:
Ok thanks :-)